### PR TITLE
libzlmediakit: add package

### PR DESCRIPTION
**Pull request by KkemChen**
libzlmediakit: add package

### CODE REVIEW

File: packages/z/zlmediakit/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package("zlmediakit")
+    set_homepage("https://github.com/ZLMediaKit/ZLMediaKit")
+    set_description("WebRTC/RTSP/RTMP/HTTP/HLS/HTTP-FLV/WebSocket-FLV/HTTP-TS/HTTP-fMP4/WebSocket-TS/WebSocket-fMP4/GB28181/SRT server and client framework based on C++11")
+    set_license("MIT")
+
+    add_urls("https://github.com/ZLMediaKit/ZLMediaKit.git",
+            "https://gitee.com/xia-chu/ZLMediaKit.git")
+
+    -- C-API
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:9]: remove comments

File: packages/z/zlmediakit/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("ENABLE_API", {description = "Enable C API SDK.", default = false, type = "boolean"})
+    add_configs("ENABLE_API_STATIC_LIB", {description = "Enable mk_api static lib.", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:11]: remove all `ENABLE_` prefix and use lower case. e.g .`api, cxx_api, webrtc ..`

File: packages/z/zlmediakit/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("ENABLE_X264", {description = "Enable libx264.", default = false, type = "boolean"})   -- Requires libx264
+
+    -- Typically no need to modify
+    add_configs("ENABLE_WEPOLL", {description = "Enable WEPOLL.", default = true, type = "boolean"})
+    add_configs("DISABLE_REPORT", {description = "Disable report to report.zlmediakit.com.", default = false, type = "boolean"})
+    add_configs("ENABLE_TESTS", {description = "Enable TESTS.", default = false, type = "boolean"})
+    add_configs("USE_SOLUTION_FOLDERS", {description = "Enable solution dir supported.", default = false, type = "boolean"})
+    add_configs("ENABLE_MSVC_MT", {description = "Enable MSVC MT.", default = true, type = "boolean"})
+    add_deps("cmake")
+    --add_links("zlmediakit","zltoolkit","flv","mpeg","srt","jsoncpp","mov","ssl","crypto")     -- The SSL version might be too low, and specific linking order may be needed.
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:47]: remove unused code

File: packages/z/zlmediakit/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+                void test() {
+                    using namespace toolkit;
+                    mINI ini;
+                    ini[".dot"] = "dot-value";
+                }
+            ]]}, {configs = {languages = "c++11"}, includes = {"Common/config.h"}}))
+        end)
+package_end()
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:107]: remove it

File: packages/z/zlmediakit/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        os.cp("**.dll", package:installdir("lib"))
+        os.cp("**.pdb", package:installdir("lib"))
+        os.cp("**.lib", package:installdir("lib"))
+
+        for _, dir in ipairs(os.dirs(ZLMediaKit_dir)) do
+            os.cp(dir,package:installdir("include"))
+        end
+
+        for _, dir in ipairs(os.dirs(ZLToolKit_dir)) do
+            os.cp(dir,package:installdir("include"))
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:91]: add spaces ` , `

File: packages/z/zlmediakit/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            "-DENABLE_WEBRTC=" .. (package:config("ENABLE_WEBRTC") and "ON" or "OFF"),
+            "-DENABLE_X264=" .. (package:config("ENABLE_X264") and "ON" or "OFF"),
+            "-DENABLE_WEPOLL=" .. (package:config("ENABLE_WEPOLL") and "ON" or "OFF"),
+            "-DDISABLE_REPORT=" .. (package:config("DISABLE_REPORT") and "ON" or "OFF"),
+            "-DUSE_SOLUTION_FOLDERS=" .. (package:config("USE_SOLUTION_FOLDERS") and "ON" or "OFF")
+        }
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        import("package.tools.cmake").install(package, configs)
+        
+        print("========================================================")
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:76]: remove logs

File: packages/z/zlmediakit/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("DISABLE_REPORT", {description = "Disable report to report.zlmediakit.com.", default = false, type = "boolean"})
+    add_configs("ENABLE_TESTS", {description = "Enable TESTS.", default = false, type = "boolean"})
+    add_configs("USE_SOLUTION_FOLDERS", {description = "Enable solution dir supported.", default = false, type = "boolean"})
+    add_configs("ENABLE_MSVC_MT", {description = "Enable MSVC MT.", default = true, type = "boolean"})
+    add_deps("cmake")
+    --add_links("zlmediakit","zltoolkit","flv","mpeg","srt","jsoncpp","mov","ssl","crypto")     -- The SSL version might be too low, and specific linking order may be needed.
+
+    on_install(function (package)
+        local configs = {
+            "-DENABLE_API=" .. (package:config("ENABLE_API") and "ON" or "OFF"),
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:51]: use for-loop

File: packages/z/zlmediakit/xmake.lua [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            "-DENABLE_FFMPEG=" .. (package:config("ENABLE_FFMPEG") and "ON" or "OFF"),
+            "-DENABLE_HLS=" .. (package:config("ENABLE_HLS") and "ON" or "OFF"),
+            "-DENABLE_JEMALLOC_STATIC=" .. (package:config("ENABLE_JEMALLOC_STATIC") and "ON" or "OFF"),
+            "-DENABLE_MEM_DEBUG=" .. (package:config("ENABLE_MEM_DEBUG") and "ON" or "OFF"),
+            "-DENABLE_MSVC_MT=" .. (package:config("ENABLE_MSVC_MT") and "ON" or "OFF"),
+            "-DENABLE_MYSQL=" .. (package:config("ENABLE_MYSQL") and "ON" or "OFF"),
+            "-DENABLE_RTPPROXY=" .. (package:config("ENABLE_RTPPROXY") and "ON" or "OFF"),
+            "-DENABLE_SERVER=" .. (package:config("ENABLE_SERVER") and "ON" or "OFF"),
+            "-DENABLE_SRT=" .. (package:config("ENABLE_SRT") and "ON" or "OFF"),
+            "-DENABLE_TESTS=" .. (package:config("ENABLE_TESTS") and "ON" or "OFF"),
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:65]: always OFF

File: packages/z/zlmediakit/xmake.lua [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+            "-DENABLE_API=" .. (package:config("ENABLE_API") and "ON" or "OFF"),
+            "-DENABLE_API_STATIC_LIB=" .. (package:config("ENABLE_API_STATIC_LIB") and "ON" or "OFF"),
+            "-DENABLE_ASAN=" .. (package:config("ENABLE_ASAN") and "ON" or "OFF"),
+            "-DENABLE_CXX_API=" .. (package:config("ENABLE_CXX_API") and "ON" or "OFF"),
+            "-DENABLE_FAAC=" .. (package:config("ENABLE_FAAC") and "ON" or "OFF"),
+            "-DENABLE_FFMPEG=" .. (package:config("ENABLE_FFMPEG") and "ON" or "OFF"),
+            "-DENABLE_HLS=" .. (package:config("ENABLE_HLS") and "ON" or "OFF"),
+            "-DENABLE_JEMALLOC_STATIC=" .. (package:config("ENABLE_JEMALLOC_STATIC") and "ON" or "OFF"),
+            "-DENABLE_MEM_DEBUG=" .. (package:config("ENABLE_MEM_DEBUG") and "ON" or "OFF"),
+            "-DENABLE_MSVC_MT=" .. (package:config("ENABLE_MSVC_MT") and "ON" or "OFF"),
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:60]: remove config, and use bultin `package:config("vs_runtime")` to handle it

File: packages/z/zlmediakit/xmake.lua [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        local ZLToolKit_dir = package:installdir("include/ZLToolKit/src/**");
+
+        os.cp("**.a", package:installdir("lib"))
+        os.cp("**.so", package:installdir("lib"))
+        os.cp("**.dll", package:installdir("lib"))
+        os.cp("**.pdb", package:installdir("lib"))
+        os.cp("**.lib", package:installdir("lib"))
+
+        for _, dir in ipairs(os.dirs(ZLMediaKit_dir)) do
+            os.cp(dir,package:installdir("include"))
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:87]: ? why?

File: packages/z/zlmediakit/xmake.lua [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+        
+        print("========================================================")
+        local ZLMediaKit_dir = package:installdir("include/ZLMediaKit/**");
+        local ZLToolKit_dir = package:installdir("include/ZLToolKit/src/**");
+
+        os.cp("**.a", package:installdir("lib"))
+        os.cp("**.so", package:installdir("lib"))
+        os.cp("**.dll", package:installdir("lib"))
+        os.cp("**.pdb", package:installdir("lib"))
+        os.cp("**.lib", package:installdir("lib"))
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:84]: os.trycp and add `package:is_plat`, package:config("shared")

File: packages/z/zlmediakit/xmake.lua [hunk 12]
```diff
@@ -0,0 +1,107 @@
+package("zlmediakit")
+    set_homepage("https://github.com/ZLMediaKit/ZLMediaKit")
+    set_description("WebRTC/RTSP/RTMP/HTTP/HLS/HTTP-FLV/WebSocket-FLV/HTTP-TS/HTTP-fMP4/WebSocket-TS/WebSocket-fMP4/GB28181/SRT server and client framework based on C++11")
+    set_license("MIT")
+
+    add_urls("https://github.com/ZLMediaKit/ZLMediaKit.git",
+            "https://gitee.com/xia-chu/ZLMediaKit.git")
+
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:8]: please add `add_versions`

File: packages/z/zlmediakit/xmake.lua [hunk 13]
```diff
@@ -0,0 +1,107 @@
+package("zlmediakit")
+    set_homepage("https://github.com/ZLMediaKit/ZLMediaKit")
+    set_description("WebRTC/RTSP/RTMP/HTTP/HLS/HTTP-FLV/WebSocket-FLV/HTTP-TS/HTTP-fMP4/WebSocket-TS/WebSocket-fMP4/GB28181/SRT server and client framework based on C++11")
+    set_license("MIT")
+
+    add_urls("https://github.com/ZLMediaKit/ZLMediaKit.git",
+            "https://gitee.com/xia-chu/ZLMediaKit.git")
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:7]: incorrect indent

File: packages/z/zlmediakit/xmake.lua [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+    add_configs("ENABLE_SERVER", {description = "ENABLE SERVER.", default = false, type = "boolean"})
+    add_configs("ENABLE_SERVER_LIB", {description = "Enable server as android static library.", default = false, type = "boolean"})
+
+    --Recommended default
+    add_configs("ENABLE_WEBRTC", {description = "ENABLE WEBRTC.", default = false, type = "boolean"})
+    add_configs("ENABLE_SCTP", {description = "Enable SCTP.", default = true, type = "boolean"})
+    add_configs("ENABLE_MP4", {description = "ENABLE_MP4.", default = true, type = "boolean"})
+    add_configs("ENABLE_HLS", {description = "ENABLE_HLS.", default = true, type = "boolean"})
+    add_configs("ENABLE_PLAYER", {description = "ENABLE PLAYER.", default = true, type = "boolean"})
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:24]: remove it, library package need not it.

File: packages/z/zlmediakit/xmake.lua [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("ENABLE_RTPPROXY", {description = "ENABLE RTPPROXY.", default = true, type = "boolean"})
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:25]: remvoe it

File: packages/z/zlmediakit/xmake.lua [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("ENABLE_OPENSSL", {description = "Enable OpenSSL.", default = false, type = "boolean"}) -- Requires openssl
+    add_configs("ENABLE_JEMALLOC_STATIC", {description = "Enable static linking to the jemalloc library.", default = false, type = "boolean"})  -- Requires jemalloc
+    add_configs("ENABLE_FFMPEG", {description = "Enable FFMPEG.", default = false, type = "boolean"})  -- Requires ffmpeg
+    add_configs("ENABLE_MYSQL", {description = "Enable MySQL.", default = false, type = "boolean"})    -- Requires mysql
+    add_configs("ENABLE_X264", {description = "Enable libx264.", default = false, type = "boolean"})   -- Requires libx264
+
+    -- Typically no need to modify
+    add_configs("ENABLE_WEPOLL", {description = "Enable WEPOLL.", default = true, type = "boolean"})
+    add_configs("DISABLE_REPORT", {description = "Disable report to report.zlmediakit.com.", default = false, type = "boolean"})
+    add_configs("ENABLE_TESTS", {description = "Enable TESTS.", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:43]: remove it

File: packages/z/zlmediakit/xmake.lua [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+    add_configs("USE_SOLUTION_FOLDERS", {description = "Enable solution dir supported.", default = false, type = "boolean"})
+    add_configs("ENABLE_MSVC_MT", {description = "Enable MSVC MT.", default = true, type = "boolean"})
```
Comment (waruqi) [inline packages/z/zlmediakit/xmake.lua:45]: add newline

File: packages/z/zlmediakit/xmake.lua (hunk 10 above)
Comment (KkemChen) [reply packages/z/zlmediakit/xmake.lua:87]: Include directory hierarchy needs to be preserved

Comment (waruqi) [reply packages/z/zlmediakit/xmake.lua:87]: os.cp support it. `os.cp ("src/**.h", "/tmp/", {rootdir="src"})`

https://xmake.io/#/manual/builtin_modules?id=oscp

**Pull request merged**